astrogrism/simulate/simulate.py

```diff
@@ -0,0 +1,90 @@
+from os import environ
+from pathlib import Path
+from shutil import copy
+from tarfile import open as tar_open
+from tempfile import gettempdir
+from warnings import warn
+
+from astropy.utils.data import download_file
+from synphot import Observation
+
+
+def generate_simulation_spectrum(grism, detector=None, temp_path=gettempdir()):
+    """
+    Initializes and uses STSynphot to generate a Vega spectrum within the bandpass of a given grism
+
+    Parameters
+    ----------
+    grism : str
+        String representation of one of the four supported HST Grisms
+        Valid grisms: G141, G102, G280, G800L
+
+    detector : int
+        For detectors with multiple chips, specifies which chip to simulate
+        Only useful for G280 and G800L Grisms
+
+    temp_path : str
+        Path to download necessary files for STSynphot. Fallsback to Python's
+        default temporary folder location
+
+    """
+    SIM_DATA_DIR = Path(temp_path) / "astrogrism_simulation_files"
+
+    if detector not in (1, 2, None):
+        raise ValueError("Invalid detector argument. Please choose 1 or 2")
+
+    # Prepare environment required for stsynphot to be imported
+    environ['PYSYN_CDBS'] = str(SIM_DATA_DIR / 'grp' / 'redcat' / 'trds')
+
+    # Download HST Instrument Data Archive
+    SIM_DATA_DIR.mkdir(parents=True, exist_ok=True)
+    hst_data_files_archive = Path(download_file('https://ssb.stsci.edu/trds/tarfiles/synphot1.tar.gz', cache=True)) # noqa
+    with tar_open(hst_data_files_archive) as tar:
+        # Only extract the files that are missing
+        for file in tar:
+            if not (SIM_DATA_DIR / Path(file.name)).exists():
+                tar.extract(file, path=SIM_DATA_DIR)
+    # Download Vega CALSPEC Reference Atlas
+    vega_reference_atlas_path = SIM_DATA_DIR / 'grp' / 'redcat' / 'trds' / 'calspec' / 'alpha_lyr_stis_010.fits' # noqa
+    # Check if it exists first before trying to download it
+    if not vega_reference_atlas_path.exists():
+        vega_reference_atlas_path.parent.mkdir(parents=True, exist_ok=True)
+        temp_download = Path(download_file('https://archive.stsci.edu/hlsps/reference-atlases/cdbs/current_calspec/alpha_lyr_stis_010.fits', cache=True)) # noqa
```

Comment:
```suggestion
        archive_url = ('https://archive.stsci.edu/hlsps/reference-atlases/cdbs/'
                       'current_calspec/alpha_lyr_stis_010.fits')
        temp_download = Path(download_file(archive_url, cache=True))
```
I strongly prefer splitting up the line with something like this to using `# noqa` on an overly long line of code. Something similar can be done with the other `noqa` lines.

Reply:
Happy to accept; I generally think of URLs as a monolithic unit, but just my opinion